### PR TITLE
prov/efa: compare pci bus id in hints

### PR DESCRIPTION
**Pull request by wzamazon**
If user specified pic bus id in hints, use it to select info.

Signed-off-by: Wei Zhang <wzam@amazon.com>

### CODE REVIEW

**Comment by wenduwan** [comment]
> If user specified pic bus id in hints, use it to select info.
> 
> Signed-off-by: Wei Zhang [wzam@amazon.com](mailto:wzam@amazon.com)

Typo in commit message `pic` -> `pci`

File: prov/efa/src/efa_prov_info.c [hunk 1]
```diff
@@ -740,3 +740,23 @@ int efa_prov_info_compare_domain_name(const struct fi_info *hints,
 
        return 0;
 }
+
+/*
+ * @brief Compare the bus id specified via hints and match it with the
+ *		  bus_id in prov_info
+ *
+ * @param      info[in]        info object
```
Comment (wenduwan) [inline prov/efa/src/efa_prov_info.c:748]: Doxygen `@param[in] info`

File: prov/efa/src/efa_prov_info.c [hunk 2]
```diff
@@ -740,3 +740,23 @@ int efa_prov_info_compare_domain_name(const struct fi_info *hints,
...
+ *             0 - No difference, names match.
+ */
+int efa_prov_info_compare_pci_bus_id(const struct fi_info *hints,
+                                     const struct fi_info *info)
+{
+	if (hints && hints->nic && hints->nic->bus_attr && hints->nic->bus_attr->bus_type == FI_BUS_PCI) {
+		return (hints->nic->bus_attr->attr.pci.bus_id == info->nic->bus_attr->attr.pci.bus_id) ? 0 : 1;
+	}
+
+	return 0;
```
Comment (wenduwan) [inline prov/efa/src/efa_prov_info.c:761]: if `hints->nic->bus_attr->bus_type` is NULL this returns 0. Is this correct?

File: prov/efa/src/efa_prov_info.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/efa_prov_info.c]: fixed

File: prov/efa/src/efa_prov_info.c (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/efa_prov_info.c:761]: `hints->nic->bus_attr->bus_type` is an integer.

I think you meant `hints->nic->bus_attr`. Yes, when it is null, we want to return 0. It means user did not specify a bus ID.

**Pull request merged**